### PR TITLE
feat(banner): add banner component

### PR DESCRIPTION
**Pull request by paul-request**
# Description

Adds a new banner component, initially with 2 variants: default and warning

Screenshot 2023-07-21 at 12 40 30

Design: https://www.figma.com/file/hJUsMXF6fNJVoa57SJuMpX/branch/x1nYeAs5DG7iMk2OMMYn2O/Components?type=design&node-id=2174-15985&mode=design&t=7ClySESHyZ8aBtQx-0

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [x] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have added the documentation
- [x] I have added any new public feature modules to public-api.ts

### CODE REVIEW

**Comment by elenagarrone** [comment]
@paul-request I've not seen a discussion for this new component 🤔 was the design approved? could you attach a link to figma please?

Also, I think the `variant` control is not working as I'm not seeing the colour being updated 😅

File: projects/canopy/src/lib/banner/docs/banner.stories.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  @Input() icon: string;
+
+  constructor(private registry: LgIconRegistry) {
+    this.registry.registerIcons(lgIconsArray);
+  }
+}
+
+// This default export determines where your story goes in the story list
+export default {
+  title: 'Components/Banner message/Examples',
```
Comment (elenagarrone) [inline projects/canopy/src/lib/banner/docs/banner.stories.ts:32]: Could this be `'Components/Banner message (WIP)/Examples'` while it's a work in progress please?

File: projects/canopy/src/lib/banner/docs/guide.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<p class="standfirst">Full-width banner to display at the top of a page, beneath the header but before the hero or other page content.</p>
+
+
+## Warning: Work in progress
+
+This component is still in development and will be enhanced with more features in the near future.
+
+Things left to do:
+
+- Add abiity to optionally dismiss the banner
```
Comment (elenagarrone) [inline projects/canopy/src/lib/banner/docs/guide.stories.mdx:17]: Small typo here

**Review by elenagarrone — DISMISSED** [review]
Nice one 👍

**Review by Xinchro — APPROVED** [review]

**Review by elenagarrone — APPROVED** [review]

**Review by Xinchro — APPROVED** [review]

**Pull request merged**